src/JBSucker.sol

```diff
@@ -650,7 +799,67 @@ abstract contract JBSucker is JBPermissioned, ERC165, IJBSucker {
     /// derived from `msg.value`)
     /// @param token The terminal token to bridge the merkle tree of.
     /// @param remoteToken The remote token which the `token` is mapped to.
-    function _sendRoot(uint256 transportPayment, address token, JBRemoteToken memory remoteToken) internal virtual;
+    function _sendRoot(uint256 transportPayment, address token, JBRemoteToken memory remoteToken) internal virtual {
+        // Ensure the token is mapped to an address on the remote chain.
+        if (remoteToken.addr == address(0)) revert JBSucker_TokenNotMapped(token);
+
+        // Make sure that the sucker still allows sending new messaged.
+        JBSuckerDeprecationState deprecationState = deprecated();
+        if (
+            deprecationState == JBSuckerDeprecationState.DEPRECATED
+                || deprecationState == JBSuckerDeprecationState.SENDING_DISABLED
+        ) {
+            revert JBSucker_Deprecated();
+        }
+
+        // Get the outbox in storage.
+        JBOutboxTree storage outbox = _outboxOf[token];
+
+        // Get the amount to send and then clear it from the outbox tree.
+        uint256 amount = outbox.balance;
+        delete outbox.balance;
+
+        // Increment the outbox tree's nonce.
+        uint64 nonce = ++outbox.nonce;
+        bytes32 root = outbox.tree.root();
+
+        uint256 count = outbox.tree.count;
+        uint256 index = count - 1;
+
+        // Update the lastestCountSend to the current count of the tree.
+        // This is used as in the fallback to allow users to withdraw locally if the bridge is reverting.
+        outbox.lastestCountSend = count;
+
+        // Emit an event for the relayers to watch for.
+        emit RootToRemote({root: root, token: token, index: index, nonce: nonce, caller: msg.sender});
+
+        // Build the message to be send.
+        JBMessageRoot memory message = JBMessageRoot({
+            token: remoteToken.addr,
+            amount: amount,
+            remoteRoot: JBInboxTreeRoot({nonce: nonce, root: root})
+        });
+
+        // Execute the chain/sucker specific logic for transferring the assets and communicating the root.
+        _sendRootOverAMB(transportPayment, index, token, amount, remoteToken, message);
+    }
+
+    function _sendRootOverAMB(
+        uint256 transportPayment,
+        uint256 index,
+        address token,
+        uint256 amount,
+        JBRemoteToken memory remoteToken,
+        JBMessageRoot memory message
+    )
+        internal
+        virtual;
+
+    /// @notice What is the maximum time it takes for a message to be received on the other side.
+    /// @dev Be sure to keep in mind if a message fails having to retry and the time it takes to retry.
+    function _maxMessagingDelay() internal view virtual returns (uint40) {
```

Comment:
can this be a constant?

Reply:
The `_maxMessagingDelay` depends on the bridge that a sucker uses, so each sucker should be able to specify how long a message might take. 14 days is just a default value thats likely safe for all suckers.